main.rst

```diff
@@ -345,17 +345,17 @@ We can then write down the gradient-adjoint-dot operator of the window
 
 The first gradient corresponds to the displacement of the source. The second gradient corresponds to
 the displacment of the destination. The third gradient corresponds to the evolution of the field.
-Usually in a particle mesh simulation, either one of the source and the destination is a fixed grid, and
+Usually in a particle mesh simulation, either one of the sources or the destination is a fixed grid, and
 the corresponding gradient vanishes.
 
 They are a bit complicated because we need to loop of the spatial dimension index :math:`a`.
-It is possible to extend these expressions to Smoothed Particle Hydrodynamics if one allow :math:`h` to be a free variable
+It is possible to extend these expressions to Smoothed Particle Hydrodynamics if one allows :math:`h` to be a free variable
 as well.
 
 Unlike the partial support of Fourier Transforms, none of the three packages we surveyed
 (TensorFlow, Theono and autograd) recognizes these resampling window operators.
 Fully implementing these operators will remove the main barrier between a generic AD software
-our cosmic initial condition problem.
+for our cosmic initial condition problem.
```

Comment:
Hmmm, this sentence does seem weird, but I am not sure what you are trying to say. Can you rewrite it and send me that version?